src/PIL/Image.py

```diff
@@ -3346,91 +3351,100 @@ def tobytes(self, offset=8):
             head = b"MM\x00\x2A\x00\x00\x00\x08"
         ifd = TiffImagePlugin.ImageFileDirectory_v2(ifh=head)
         for tag, value in self.items():
+            if tag in [0x8769, 0x8225] and not isinstance(value, dict):
+                value = self.get_ifd(tag)
```

Comment:
This does not work for interop since it's inside of Exif IFD. 
```
im = Image.open('10_years_of_Wikipedia_by_Guillaume_Paumier.jpg')
good_exif = im.getexif()

bad_exif = Exif()
bad_exif.load(good_exif.tobytes())

print("Good interop dictionary size: %d" % len(good_exif.get_ifd(0xA005)))
print(" Bad interop dictionary size: %d" % len(bad_exif.get_ifd(0xA005)))

```
I suspect same for makernote too, but since makernote is vendor specific, I think it would be safer to make it read only property.

Reply:
Ok, I've added another commit to cover Interop.

Reply:
I've being thinking through this on and off for some time.
Thinking of consumer of this class, it would make sense when reading Exif IFD to get Interop value decoded into a dict at that time. Or shimmed with `ImageFileDirectory_v2` instance. Otherwise caller must know Inerop tag ID, must know to call `get_ifd`, etc. So that's a bit inconvenient and adds extra knowledge to consumer. 
Same for Makernote, however saving it back we don't want to encode it back to vendor-specific format and probably worth just copying whatever bytes were there originally.
On the other hand Exif is quite complicated and some knowledge from consumer is expected anyway.
:shrug: 
What do you think of this?

Reply:
Another benefit of using `ImageFileDirectory_v2` as a shim for sub-IFDs is that we get lazy loading.